proxy_network.c

```diff
@@ -1306,10 +1300,15 @@ static bool _post_pending_write(mcp_backend_t *be, ssize_t sent) {
         if (flushed) {
             did_flush = flushed;
         }
+        if (sent <= 0) {
+            // really shouldn't be negative, though.
+            assert(sent >= 0);
+            break;
+        }
     } // for
 
     // resume the flush from this point.
-    if (io != NULL) {
+    if (io != NULL && !io->flushed) {
```

Comment:
This makes sense.

Wonder if the benefit is to remove redundant loop by adding this check?

Reply:
The whole loop needs to be redone; when there are separate lists for to-read and to-write none of that logic will remain. This PR is attempting a minimal churn in the meantime.